packages/plugin-ext/src/plugin/scm.ts

```diff
@@ -38,17 +39,23 @@ import { RPCProtocol } from '../common/rpc-protocol';
 import { URI } from './types-impl';
 import { ScmCommandArg } from '../common/plugin-api-rpc';
 import { sep } from '@theia/core/lib/common/paths';
+import { ThemeIcon } from '@theia/monaco-editor-core/esm/vs/platform/theme/common/themeService';
+import { PluginIconPath } from './plugin-icon-path';
 type ProviderHandle = number;
 type GroupHandle = number;
 type ResourceStateHandle = number;
 
-function getIconResource(decorations?: theia.SourceControlResourceThemableDecorations): theia.Uri | undefined {
+function getIconResource(decorations?: theia.SourceControlResourceThemableDecorations): theia.Uri | ThemeIcon | undefined {
     if (!decorations) {
         return undefined;
     } else if (typeof decorations.iconPath === 'string') {
         return URI.file(decorations.iconPath);
-    } else {
+    } else if (URI.isUri(decorations.iconPath)) {
+        return decorations.iconPath;
+    } else if (ThemeIcon.isThemeIcon(decorations.iconPath)) {
         return decorations.iconPath;
+    } else {
```

Comment:
Isn't this an error condition that should be logged as a warning?

Reply:
Thanks!
Will do.

Reply:
Currently there are two possible arguments for the `decorations` are of type `undefined` or `SourceControlResourceThemableDecorations`.

Then the function mostly cares about the `iconPath` attribute that can be `undefined` or of type `ThemeIcon`, `URI` or `string`. 

The case where `decorations` is not defined is covered by the first `if` statement. This also covers the case where `iconPath` is not defined within `decorations`. This is used mainly to determine if `decorations` has a `light` and `dark` icons. The other cases for `iconPath` being a URI, ThemeIcon or string are also accounted for, That last `else` is there for unexpected situations. I don't expect the warning to be called at all.

However, if a proper warning message should be placed then I think it is probably better to go for the value type instead of the argument. I am not sure how to do that though. For now, I'll add the value of `iconPath` to the message.